Jenkinsfile

```diff
@@ -122,8 +122,10 @@ pipeline {
                                                        excludeFile("_build\\.external\\/.*")]
                             }
                         }
-                        success {
-                            sh "rm -rf _build.external"
+                        cleanup {
+                            dir('_build.external') {
+                                deleteDir()
+                            }
```

Comment:
This shouldn't be necessary as it's done in [`sconsBuild`](https://github.com/daos-stack/pipeline-lib/blob/master/vars/sconsBuild.groovy#L173) when passed in the [`clean:` argument above](https://github.com/daos-stack/iof/pull/25/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R110).

It was decided to do it this way to leave the externals build tree in-tact for post-build analysis if necessary.

Reply:
Ok, I was just changing code from sh("rm -rf../") to make it clearer and consistent but I can remove entirely.

Reply:
Right, and looking at it again, I see that we are actually inconsistent between daos and cart in this regard so there goes my consistency argument.

I do recall adding this to the daos pipeline to keep transient space usage under control because `_build.externals/` is a very significant space consumer -- as much as I would actually prefer the stance of keeping it around for post-build inspection.

Given the fewer dependencies for cart and iof, it's less so and as we migrate towards RPMs, it will be even less so.

So, my apologies for my wishy-washiness on this.  Since it's already inconsistent I won't mind which way this PR goes in that regard.

Reply:
At this point, I only really have two requirements for this,

* Be able to run IOF tests against CaRT, preferably as part of the PR process
* CaRT and DOAS should be consistent.  Having worked to integrate the IOF and DOAS repos there are a lot of differences between then that aren't easily explained and are causing me lots of headaches.